src/System.Net.Security/tests/FunctionalTests/SslStreamStreamToStreamTest.cs

```diff
@@ -71,6 +63,79 @@ public void SslStream_StreamToStream_Authentication_IncorrectServerName_Fail()
             }
         }
 
+        [Fact]
+        public void SslStream_StreamToStream_Back2Back_ClientWrite_Success()
+        {
+            SslStream clientSSLStream = null;
+            SslStream ServerSSLStream = null;
+
+            try
+            {
+                GetServerClientSSLStream(out clientSSLStream, out ServerSSLStream);
+                clientSSLStream.Write(sampleMsg);
+                clientSSLStream.Write(sampleMsg);
+            }
+            finally
+            {
+                DisposeServerClientSSLStream(clientSSLStream, ServerSSLStream);
+            }
+        }
+
+        #endregion
+
+        #region Private Methods
+
+        private bool AllowAnyServerCertificate(
+            object sender,
+            X509Certificate certificate,
+            X509Chain chain,
+            SslPolicyErrors sslPolicyErrors)
+        {
+            if (sslPolicyErrors != SslPolicyErrors.RemoteCertificateNameMismatch)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private void GetServerClientSSLStream(out SslStream clientSSLStream, out SslStream ServerSSLStream)
+        {
+            MockNetwork network = new MockNetwork();
+
+            var clientStream = new FakeNetworkStream(false, network);
+            var serverStream = new FakeNetworkStream(true, network);
+
+            clientSSLStream = new SslStream(clientStream, false, AllowAnyServerCertificate);
+            ServerSSLStream = new SslStream(serverStream);
+
+            X509Certificate2 certificate = TestConfiguration.GetServerCertificate();
+            Task[] auth = new Task[2];
+
+            auth[0] = clientSSLStream.AuthenticateAsClientAsync(certificate.Subject);
```

Comment:
This would be better as certificate.GetNameInfo(X509NameType.SimpleName, false).  Then it'd be the CN value from the subject (if present) vs the whole subject line (which is a guaranteed non-match)